### ISSUE TITLE
error: 'utf-8' codec can't decode bytes

### ISSUE THREAD

**Issue by Andy2244** [adameste/anidbcli#24]
Getting those errors for a [batch](https://anidb.net/anime/14449/?gid=15782#eplist), but only episode files 8,12, give this error?
```
[INFO] Processing file "d:\[Judas] Hamefura - S01E08.mkv"
[SUCCESS] Generated ed2k link.
[ERROR] Failed to get file info: 'utf-8' codec can't decode bytes in position 1397-1398: invalid continuation byte
[INFO] Processing file "d:\[Judas] Hamefura - S01E12.mkv"
[SUCCESS] Generated ed2k link.
[ERROR] Failed to get file info: 'utf-8' codec can't decode byte 0xe3 in position 1398: invalid continuation byte
```

**Comment by Andy2244** [comment]
Full log:
```
[INFO] Processing file "d:\[Judas] Hamefura - S01E08.mkv"
[SUCCESS] Generated ed2k link.
Traceback (most recent call last):
  File "c:\python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Python38\Scripts\anidbcli.exe\__main__.py", line 7, in <module>
  File "c:\python38\lib\site-packages\anidbcli\cli.py", line 143, in main
    cli(obj={})
  File "c:\python38\lib\site-packages\click\core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "c:\python38\lib\site-packages\click\core.py", line 782, in main
    rv = self.invoke(ctx)
  File "c:\python38\lib\site-packages\click\core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "c:\python38\lib\site-packages\click\core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "c:\python38\lib\site-packages\click\core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "c:\python38\lib\site-packages\click\decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "c:\python38\lib\site-packages\anidbcli\cli.py", line 88, in api
    res = operation.Process(file_obj)
  File "c:\python38\lib\site-packages\anidbcli\operations.py", line 91, in Process
    res = self.connector.send_request(API_ENDPOINT_FILE % (file["size"], file["ed2k"]))
  File "c:\python38\lib\site-packages\anidbcli\anidbconnector.py", line 117, in send_request
    res = self.crypto.Decrypt(res)
  File "c:\python38\lib\site-packages\anidbcli\encryptors.py", line 14, in Decrypt
    return message.decode("utf-8")
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 1397-1398: invalid continuation byte
```

**Comment by Andy2244** [comment]
> The episodes may have been corrupted while downloading, please try redownloading them too.

I already use the latest python3 _(3.8.5)_ and pip _(20.2.2)_ and my local file hashes `(crc32 + ED2K)` also match the "upstream" aka https://animetosho.org/file/judas-hamefura-s01e08-mkv.728927 . 

So there is no file corruption or version problem, so can you post your versions and check your hashes and see if its the same file we are talking about?

**Comment by kodomono** [comment]
Yes, we are talking about the same file. I used https://nyaa.si/view/1257030.

```
[INFO] Processing file "/home/kuro/Downloads/[Judas] Hamefura (Season 1) [1080p][HEVC x265 10bit][Multi-Subs]/[Judas] Hamefura - S01E12.mkv"
[SUCCESS] Generated ed2k link.
[INFO] ed2k://|file|[Judas] Hamefura - S01E12.mkv|322691214|6661ad147e864af035359ad2c518197b|
[SUCCESS] Mylist entry added.
[INFO] Processing file "/home/kuro/Downloads/[Judas] Hamefura (Season 1) [1080p][HEVC x265 10bit][Multi-Subs]/[Judas] Hamefura - S01E08.mkv"
[SUCCESS] Generated ed2k link.
[INFO] ed2k://|file|[Judas] Hamefura - S01E08.mkv|514700408|36c8d47c6845ea085b25f724e7736763|
[SUCCESS] Mylist entry added.
```
which also matches with [aniDB](https://anidb.net/f2549420) and [animetosho](https://animetosho.org/file/judas-hamefura-s01e12-mkv.728920).

This is a weird error. I doubt it occurs because of an UnicodeDecodeError.
This is just a guess, but maybe editing [this line](https://github.com/adameste/anidbcli/blob/master/anidbcli/encryptors.py#L14) to something like `return message.decode("utf-8",  errors="igonore")` or `return message.decode("utf-8",  errors="replace")` and then reinstalling it does the trick?

**Comment by Andy2244** [comment]
> return message.decode("utf-8", errors="replace")

ok, yes this actually did the trick. Will setup a PR for it thanks, guess its [this](https://wiki.python.org/moin/UnicodeDecodeError).

**Comment by kodomono** [comment]
Oh, that's nice!

**Comment by adameste** [comment]
Sorry, I have quite a few other things to do, fix is included in the new release (1.66) sou you can just update the plugin via ```pip install --upgrade anidbcli```
